src/wp-includes/load.php

```diff
@@ -607,7 +607,16 @@ function wp_debug_mode() {
 	}
 
 	if ( WP_DEBUG ) {
-		error_reporting( E_ALL );
+		if ( WP_DEBUG_LEVEL ) {
+
+			$wp_debug_level = WP_DEBUG_LEVEL;
+			// In the main function:
+			if ( ! wp_is_valid_error_level( $wp_debug_level ) ) {
+				$wp_debug_level = E_ALL;
+			}
+
+			error_reporting( $wp_debug_level );
+		}
```

Comment:
Logically I think we're also missing the else here? 
```suggestion
		} else {
		error_reporting( E_ALL );
		}
```

Or I think we can end up with no `error_reporting` set at all. You can compare to my previous PR too where I did this.